Tests/AppTests/AnalyzerVersionThrottlingTests.swift

```diff
@@ -304,6 +305,7 @@ class AnalyzerVersionThrottlingTests: AppTestCase {
 
             // validate
             XCTAssertEqual(res, [new])
+            // XCTAsserEqual(res, [old]) ??
         }
```

Comment:
This looks weird - note that `new` is actually the older version and we choose it, because if the old (or rather: existing) version is outside the window we indiscriminately choose the incoming version. There is no check if it's actually newer.

I _think_ this is still the correct thing to do, because it reflects the scenario where a branch is force pushed to an older revision.

Reply:
This is a complicated situation. The force pushing would need to be within the 24-hour window for that scenario you mentioned to affect this.

But, this test as it's written, I'd expect the assert to be `[old]` and not `[new]`, so it's potentially concerning that this passes as-is.

Reply:
It's actually the other way around - it's outside the 24h window and if we didn't do what we currently do we'd fail to purge the outdated version forever.

In other words, I think we _must_ do what we're currently doing or a force push never get the intended version through.

Should add that as a comment to the test to clarify.

Reply:
I've renamed the variables and the test to make this clearer.

Reply:
As we discussed, I tripped up on the meaning of the `ex` variable. My mind kept going to `excluded` rather than `existing`. I'd personally rename them to `existing` and `incoming` for extra clarity but I'll leave it to you.